### PR TITLE
Adds README and tests

### PR DESCRIPTION
**Pull request by ezmiller**
@thomastuts @AnnJohn  opening this lab for initial review. It may be a bit more rough than usual, as I went back and forth on a few things as I was writing it. TBut I'm opening it now because I think I could benefit from other eyes taking a look. The difference in this lab is that I constructed it as an exercise in refactoring, which I think is an excellent way to teach the idea of pushing state to the edges because it makes in concrete and also emphasizes that this is a **design choice** or matter of discipline that is easily ignored. Thomas do you think the tests are setup in such a way as to nudge students in the right direction. I tried to give the student guidance in the last "How to Approach This Lab" section. Looking forward to both of your input.

### CODE REVIEW

**Comment by ezmiller** [comment]
@thomastuts @AnnJohn as I'm working on the next lessons in unit 6, I'm realizing that this lab may possibly work better as the final lab in the unit. I'm still thinking this through as I write, but might be good to keep that in mind when doing the initial review. 

What I'm thinking is that it might be good to move this lab and the lesson on props with which it is paired to the end. So the order of the unit would be presentation components -> container components -> (and then) passing props/pushing state to the edges of the app...

**Comment by thomastuts** [comment]
@ezmiller I still have to go through the content of this particular lab, but I think an argument can be made for both options. I think the current order would work too, since you have a natural progression of writing components:
1. Components that contain a lot of props (presentational/dumb components)
2. Components that contain a lot of state on the top level and pass those props down (these components could still use DOM markup, since they're not container components just _quite_ yet)
3. Container components that are only responsible for rendering dumb components and no DOM themselves (so actual _smart_ components)

I feel like moving props to the edge of the application is sort of a prelude towards thinking about container components — they're _sort of_ the same thing, which is why I think both options are viable. WDYT?

**Comment by ezmiller** [comment]
@thomastuts what you say makes sense thanks. I hadn't seen that particular development of the ideas. 

What I focused on was the way the concept of pushing state to the edge of the application invokes the bigger context of the entire application. It seemed awkward as I moved through that material to go from individual dumb components, to the discussion of the application context, and then again down to a self-contained example of container components. 

I think my thinking was also influenced by the fact that the last three lessons in the spreadsheet include the objective of talking about moving stage to the edge of an application, something that I covered already in the passing-props lessons. I worried that came too early... 

In any case, maybe we can discuss again later, and keep this in mind while we go through the review. It can all be reorganized later anyway I suppose.

File: README.md [hunk 1]
```diff
@@ -5,13 +5,82 @@
...
+But the bottom line is that this app will help people get the fruit that they
+truly want!
+
+![](https://media.giphy.com/media/97ZWlB7ENlalq/giphy.gif)
+
+## The Task at Hand
+
+Now, one thing that is a bit unusual about this lab is that rather than building
+the app from scratch, we will instead work with an application that has already
+been built. If you run `npm run budle` and then load the `index.html` file,
```
Comment (thomastuts) [inline README.md]: `npm run budle` -> `npm run bundle`

**Review by thomastuts — CHANGES_REQUESTED** [review]
This looks amazing, @ezmiller! Instructions are super clear, I think students will really enjoy this lab. There's one really minor typo, as well as some JSX statements that use 4-space indentation. When those are fixed, this is good to merge! :shipit:

**Comment by ezmiller** [comment]
Thanks @thomastuts! I fixed the typo. Can you indicate where you are seeing the spacing problems with the JSX? I couldn't find it.

File: components/App.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const React = require('react');
+
+const FruitBasket = require('./FruitBasket');
+
+const App = () => {
+  return (
+		<FruitBasket />
```
Comment (thomastuts) [inline components/App.js]: 2-space indentation

File: components/FilteredFruitList.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      .then(res => res.json())
+      .then(fruit => this.setState({ items: fruit}));
+  }
+
+  render() {
+    const list = !this.props.filter ? this.state.items :
+      this.state.items.filter(i => i.fruit_type == this.props.filter);
+
+		return (
+			<ul className="fruit-list">
```
Comment (thomastuts) [inline components/FilteredFruitList.js]: 2-space indentation

**Review by thomastuts — CHANGES_REQUESTED** [review]
@ezmiller Marked them for you! I thought there were more, but there are only 2 places where the indentation is off (it looks like it's using a tab character instead of 4 spaces)

**Comment by thomastuts** [comment]
@ezmiller The `package.json` is also missing the `sinon`, `fetch-mock` and `emojilib` dependencies, the tests are failing right now on both branches.

**Comment by ezmiller** [comment]
@thomastuts the linter tests are still failing here. do you know why that might be?

**Comment by AnnJohn** [comment]
@ezmiller don't worry about the linter tests. you can feel free to merge if/when this is ready. thanks!

**Pull request merged**